multiapps-controller-process/src/test/java/org/cloudfoundry/multiapps/controller/process/steps/UnbindServiceStepFromApplicationTest.java

```diff
@@ -1,25 +1,67 @@
 package org.cloudfoundry.multiapps.controller.process.steps;
 
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.verify;
 
 import org.cloudfoundry.multiapps.controller.client.lib.domain.CloudApplicationExtended;
 import org.cloudfoundry.multiapps.controller.client.lib.domain.ImmutableCloudApplicationExtended;
 import org.cloudfoundry.multiapps.controller.process.variables.Variables;
 import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.http.HttpStatus;
+
+import com.sap.cloudfoundry.client.facade.ApplicationServicesUpdateCallback;
+import com.sap.cloudfoundry.client.facade.CloudOperationException;
+import com.sap.cloudfoundry.client.facade.domain.CloudServiceBinding;
 
 class UnbindServiceStepFromApplicationTest extends SyncFlowableStepTest<UnbindServiceFromApplicationStep> {
 
+    @Mock
+    private CloudServiceBinding cloudServiceBinding;
+
     private static final String APPLICATION_NAME = "test_application";
     private static final String SERVICE_NAME = "test_service";
 
     @Test
     void testUnbindServiceStep() {
         prepareContext();
 
+        Mockito.when(client.getServiceBindingForApplication(any(), any()))
+               .thenReturn(cloudServiceBinding);
         step.execute(execution);
 
         assertStepFinishedSuccessfully();
-        verify(client).unbindServiceInstance(APPLICATION_NAME, SERVICE_NAME);
+        verify(client).unbindServiceInstance(eq(APPLICATION_NAME), eq(SERVICE_NAME), any(ApplicationServicesUpdateCallback.class));
+        verify(client, never()).getServiceBindingForApplication(any(), any());
+    }
+
+    @Test
+    void testDoNotThrowExceptionWhenServiceBindingAlreadyDeleted() {
+        ProcessContext customProcessContext = new ProcessContext(execution, stepLogger, clientProvider);
+        Mockito.when(client.getServiceBindingForApplication(any(), any()))
+               .thenReturn(null);
+
+        assertDoesNotThrow(() -> handleErrorInCallback(customProcessContext));
+    }
+
+    @Test
+    void testThrowExceptionWhenBindingExist() {
+        ProcessContext customProcessContext = new ProcessContext(execution, stepLogger, clientProvider);
+        Mockito.when(client.getServiceBindingForApplication(any(), any()))
+               .thenReturn(cloudServiceBinding);
+
+        assertThrows(CloudOperationException.class, () -> handleErrorInCallback(customProcessContext));
```

Comment:
Verify that the returned code is 502

Reply:
I think it is not needed, we will just re-throw it.